saw-core/src/Verifier/SAW/Simulator/Prims.hs

```diff
@@ -672,18 +678,20 @@ updOp bp =
         | toInteger i < toInteger (V.length xv)
            -> return (VVector (xv V.// [(fromIntegral i, y)]))
         | otherwise                   -> return (VVector xv)
-      VToNat (VWord w) ->
-        do let wsize = bpBvSize bp w
-               f i = do b <- bpBvEq bp w =<< bpBvLit bp wsize (toInteger i)
+      VBVToNat wsize (VWord w) ->
+        do let f i = do b <- bpBvEq bp w =<< bpBvLit bp wsize (toInteger i)
                         if wsize < 64 && toInteger i >= 2 ^ wsize
                           then return (xv V.! i)
                           else delay (lazyMuxValue bp b (force y) (force (xv V.! i)))
            yv <- V.generateM (V.length xv) f
            return (VVector yv)
-      VToNat (VVector iv) ->
+      VBVToNat _sz (VVector iv) ->
         do let update i = return (VVector (xv V.// [(i, y)]))
            iv' <- V.mapM (liftM toBool . force) iv
            selectV (lazyMuxValue bp) (fromIntegral n - 1) update iv' -- FIXME dangerous fromIntegral
+
+      VIntToNat _ -> error "updOp: symbolic integer TODO"
```

Comment:
A symbolic-integer-index update should be implementable by putting an equality-test if-then-else in each array position